### ISSUE TITLE
Release lintr 3.2.0

### ISSUE THREAD

**Issue by MichaelChirico** [r-lib/lintr#2392]
Glanced at the NEWS and there's plenty there already for us to ship off to CRAN.

Prepare for release:

* [x] [`3.2.0` milestone](https://github.com/r-lib/lintr/milestone/7)
* [x] Re-read and refine the entire NEWS for the release
* [x] Ensure `backport_linter()` covers all released R versions
* [x] Check URLs: `urlchecker::url_check()` or `tools:::check_url_db(tools:::url_db_from_package_sources("."))`
* [x] Run the `R-CMD-check` GHA on _all_ released R versions on Ubuntu, Mac, and Windows. See #2501 for reference
* [x] `devtools::check(remote = TRUE, manual = TRUE, force_suggests = TRUE)`
* [x] `devtools::check_win_devel()` or `devtools::check_win_devel(email = "your@email")`
* <skipped> #2720
* [x] Run [`revdepcheck::revdep_check(dependencies = c("Depends", "Imports"))`](https://revdepcheck.r-lib.org/reference/revdep_check.html)
* <skipped> Run `.dev/revdep_get_repos.R` and `.dev/revdep_compare_releases.R`

Submit to CRAN:

* [x] `usethis::use_version('patch')`
* [ ] `devtools::submit_cran()`
* [x] Approve email

Wait for CRAN...

* [ ] Update `cran-comments.md`
* [x] Accepted :tada:
* [ ] `usethis::use_github_release()`
* [ ] `usethis::use_dev_version()`
* [ ] Toot/Tweet

**Comment by IndrajeetPatil** [comment]
> (PS, do we think it may be time for 3.2.0, esp. given all the new linters we've added?)

Oh, absolutely!

**Comment by MichaelChirico** [comment]
OK, renaming the milestones now.

**Comment by MichaelChirico** [comment]
@AshesITR with the CRAN failure (#2527), let's decide how to proceed. Options I see:

 1. Quick patch release -- branch from 3.1.1, fix only #2527, release to CRAN as 3.1.2 (also mirror the change on `main`)
 2. Finish up the last two milestone (https://github.com/r-lib/lintr/milestone/7) issues: #2431 (related PR: #2496), #2325.
 3. Delay #2431 by reverting #2357 for this release
 4. Delay #2431 by making note in the NEWS that #2357 is experimental/subject to change in the next release
 
#2325 is also not strictly required for this release, but we have the implementation ready & just need to decide on argument naming/defaults IIRC.

Let me know your thoughts based on your availability in the next two weeks or so.

**Comment by AshesITR** [comment]
I won't be able to contribute anything meaningful in the upcoming weeks unfortunately.

My hunch is option 1 (maybe including the maintainer change) is the least risky, but I trust your judgement.

**Comment by MichaelChirico** [comment]
SGTM! FYI (cc also @IndrajeetPatil), I don't plan any major change from taking over maintainership, I just think of it as passing responsibility for the actual CRAN submission process from Jim to me. GH review/etc process has been working pretty well in general.

**Comment by MichaelChirico** [comment]
Yes, we're absolutely on the precipice of release. But I'm stuck on #2431, there's a bit of a design decision there that would be harder to change after release & haven't found the time to come to a proper conclusion there yet. Opinions+insights welcome.

Or, we can just mark that as "experimental/likely to change" and go ahead and release.

**Comment by MichaelChirico** [comment]
OK, finally got the email from Brian about {patrick} update breaking us. Update due in 3 weeks or so.

**Comment by MichaelChirico** [comment]
OK, only `revdepcheck::revdep_check()` left in the release process, everything else looks good to go. There were a few minor edits required: all is on https://github.com/r-lib/lintr/tree/edits-3.2.0 for now, will propose it as a PR once it's accepted to CRAN.

**Comment by MichaelChirico** [comment]
now on CRAN! 🎉 thanks everyone :)

cc @IndrajeetPatil I think this unblocks the JOSS paper

**Comment by IndrajeetPatil** [comment]
Thanks a lot for taking care of this, Michael! 😊

I will comment in the JOSS thread, alerting reviewers.
